src/Storages/Statistics/StatisticsTDigest.cpp

```diff
@@ -16,12 +17,16 @@ StatisticsTDigest::StatisticsTDigest(const SingleStatisticsDescription & stat_)
 void StatisticsTDigest::update(const ColumnPtr & column)
 {
     size_t rows = column->size();
-
     for (size_t row = 0; row < rows; ++row)
     {
-        /// TODO: support more types.
-        Float64 value = column->getFloat64(row);
-        t_digest.add(value, 1);
+        Field f;
+        column->get(row, f);
+
+        if (f.isNull())
+            continue;
+
+        if (auto float_val = IStatistics::getFloat64(f))
```

Comment:
`auto field_as_float = [...]`

Reply:
I guess we should throw an exception if the casted value is not a Float64.

Reply:
I tried to supports data type `String` in statistics `count_min`